DashSync/shared/Models/Managers/Chain Managers/DSChainManager.m

```diff
@@ -346,10 +346,8 @@ - (double)combinedSyncProgress {
     DSLog(@"combinedSyncProgress breakdown %f %f %f", self.terminalHeaderSyncProgress, self.masternodeManager.masternodeListAndQuorumsSyncProgress, self.chainSyncProgress);
 #endif
     if ((self.terminalHeaderSyncWeight + self.chainSyncWeight + self.masternodeListSyncWeight) == 0) {
-        if (self.peerManager.connected) {
-            return 1;
-        } else {
-            return 0;
+        @synchronized (self.peerManager) {
```

Comment:
@pankcuf I don't think `@synchronized` is a real solution, I personally against using `@synchronized` because it can lead performance decreasing and other "_unknowns_"